java/integTest/src/test/java/glide/cluster/CommandTests.java

```diff
@@ -3014,4 +3018,308 @@ public void test_cluster_scan_all_stream() {
+                        .get();
+        assertArrayEquals(expected, result);
+        assertArrayEquals(expected, result2);
+    }
+
+    @Test
+    @SneakyThrows
+    public void scriptExistsBinary() {
+        Script script1 = new Script(gs("return 'Hello'"), true);
+        Script script2 = new Script(gs("return 'World'"), true);
+        Script script3 = new Script(gs("return 'Hello World'"), true);
+        String key = UUID.randomUUID().toString();
+        SingleNodeRoute route = new SlotKeyRoute(key, PRIMARY);
+        Boolean[] expected = new Boolean[] {true, false, true, false};
+
+        // Load script1 to all nodes, do not load script2 and load script3 with route
+        clusterClient.invokeScript(script1).get();
+        clusterClient.invokeScript(script3, route).get();
+
+        // Get the SHA1 digests of the scripts
+        GlideString sha1_1 = gs(script1.getHash());
+        GlideString sha1_2 = gs(script2.getHash());
+        GlideString sha1_3 = gs(script3.getHash());
+        GlideString nonExistentSha1 = gs("0".repeat(40)); // A SHA1 that doesn't exist
+
+        // Check existence of scripts
+        Boolean[] result =
+                clusterClient
+                        .scriptExists(new GlideString[] {sha1_1, sha1_2, sha1_3, nonExistentSha1})
+                        .get();
+        Boolean[] result2 =
+                clusterClient
+                        .scriptExists(new GlideString[] {sha1_1, sha1_2, sha1_3, nonExistentSha1}, route)
+                        .get();
+        assertArrayEquals(expected, result);
+        assertArrayEquals(expected, result2);
+    }
+
+    @Test
+    @SneakyThrows
+    public void scriptFlush() {
+        Script script = new Script("return 'Hello'", true);
+
+        // Load script
+        clusterClient.invokeScript(script, ALL_PRIMARIES).get();
+
+        // Check existence of scripts
+        Boolean[] result =
+                clusterClient.scriptExists(new String[] {script.getHash()}, ALL_PRIMARIES).get();
+        assertArrayEquals(new Boolean[] {true}, result);
+
+        // flush the script cache
+        assertEquals(OK, clusterClient.scriptFlush(ALL_PRIMARIES).get());
+
+        // check that the script no longer exists
+        result = clusterClient.scriptExists(new String[] {script.getHash()}, ALL_PRIMARIES).get();
+        assertArrayEquals(new Boolean[] {false}, result);
+
+        // Test with ASYNC mode
+        clusterClient.invokeScript(script, ALL_PRIMARIES).get();
+        assertEquals(OK, clusterClient.scriptFlush(FlushMode.ASYNC, ALL_PRIMARIES).get());
+        result = clusterClient.scriptExists(new String[] {script.getHash()}, ALL_PRIMARIES).get();
+        assertArrayEquals(new Boolean[] {false}, result);
+    }
+
+    @Test
+    @SneakyThrows
+    public void scriptKill_with_route() {
+        // create and load a long-running script and a primary node route
+        Script script = new Script(createLongRunningLuaScript(10, true), true);
```

Comment:
```suggestion
        Script script = new Script(createLongRunningLuaScript(7, true), true);
```
Script timeout >= test timeout, so in case of failure or delay test won't be able to clean up (wait for script)

Update in all places

Reply:
I use 5 for Node's `script` tests.